azurerm/resource_arm_network_interface_test.go

```diff
@@ -37,16 +38,133 @@ func TestAccAzureRMNetworkInterface_disappears(t *testing.T) {
 			{
 				Config: testAccAzureRMNetworkInterface_basic(rInt, testLocation()),
 				Check: resource.ComposeTestCheckFunc(
-					testCheckAzureRMNetworkInterfaceExists("azurerm_network_interface.test"),
-					testCheckAzureRMNetworkInterfaceDisappears("azurerm_network_interface.test"),
+					testCheckAzureRMNetworkInterfaceExists(resourceName),
+					testCheckAzureRMNetworkInterfaceDisappears(resourceName),
 				),
 				ExpectNonEmptyPlan: true,
 			},
 		},
 	})
 }
 
+func TestAccAzureRMNetworkInterface_setNetworkSecurityGroupId(t *testing.T) {
+	resourceName := "azurerm_network_interface.test"
+	rInt := acctest.RandInt()
+	location := testLocation()
+	config := testAccAzureRMNetworkInterface_basic(rInt, location)
+	updatedConfig := testAccAzureRMNetworkInterface_basicWithNetworkSecurityGroup(rInt, location)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNetworkInterfaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetworkInterfaceExists(resourceName),
+				),
+			},
+			{
+				Config: updatedConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetworkInterfaceExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "network_security_group_id"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMNetworkInterface_removeNetworkSecurityGroupId(t *testing.T) {
+	resourceName := "azurerm_network_interface.test"
+	rInt := acctest.RandInt()
+	location := testLocation()
+	config := testAccAzureRMNetworkInterface_basicWithNetworkSecurityGroup(rInt, location)
+	updatedConfig := testAccAzureRMNetworkInterface_basic(rInt, location)
```

Comment:
Nitpick, but I think `config` & `updatedConfig` don't need to be variables for that single purpose a few lines below. 🙂